### DISCUSSION TITLE
A question about Cue language specification

### DISCUSSION THREAD

**Discussion by quovadis-7** [cue-lang/cue#1333] [Q&A]
When I browse the content of this page：https://cuelang.org/docs/references/spec/#import-declarations
I am a little confused about "ImportPath" for the use:
ImportPath       = `"` ImportLocation [ ":" identifier ] `"`
In which case I should use " [ ":" identifier ] "?
When I write a ImportPath as follow:
s "al/aone-oss/serverless-iac/serverless"
In the PsiViewer, it doesn't parse out the colon and the identifier after the colon, gitlab.alibaba-inc.com/test/server/serverless:serv is parsed as a  UNICODE_VALUE.
![image](https://user-images.githubusercontent.com/22310113/134910349-f3f408ed-f22d-4b93-b85e-57d9cab15041.png)

**Comment by eonpatapon** [comment]
Import path is `path/to/module:package`, however if the package has the same name as the module you can omit it

**Comment by quovadis-7** [comment]
> Import path is `path/to/module:package`, however if the package has the same name as the module you can omit it

Can you explain what is a module and can you show me an concrete instance for the use of ImportPath = " ImportLocation  ":" identifier "

**Comment by eonpatapon** [comment]
Actually the import path is a path to the package you want to import (not the module). But the last component of the path might be different from the actual package name specified in cue files.

This section of the doc explains it: https://cuelang.org/docs/concepts/packages/#import-path

For example I have a project layout like this:

```
myProject/
  cue.mod/
    module.cue
  packageX/
    main.cue
  y/
    main.cue
```

`packageX/main.cue`:

```cue
package x

x: true
```

`y/main.cue`:

```cue
package y

y: true
```

Now if I want to import the `y` package I can simply do `import "github.com/org/myProject/y"` because the last part of the package path match the package name defined in the cue file. Whereas to import package `x` I need to do `import "github.com/org/myProject/packageX:x"`

Also, the `x`, `y` are part of a module being `github.com/org/myProject`. The module name should be referenced in `cue.mod/module.cue`.

**Comment by quovadis-7** [comment]
> Actually the import path is a path to the package you want to import (not the module). But the last component of the path might be different from the actual package name specified in cue files.
> 
> This section of the doc explains it: https://cuelang.org/docs/concepts/packages/#import-path
> 
> For example I have a project layout like this:
> 
> ```
> myProject/
>   cue.mod/
>     module.cue
>   packageX/
>     main.cue
>   y/
>     main.cue
> ```
> 
> `packageX/main.cue`:
> 
> ```
> package x
> 
> x: true
> ```
> 
> `y/main.cue`:
> 
> ```
> package y
> 
> y: true
> ```
> 
> Now if I want to import the `y` package I can simply do `import "github.com/org/myProject/y"` because the last part of the package path match the package name defined in the cue file. Whereas to import package `x` I need to do `import "github.com/org/myProject/packageX:x"`
> 
> Also, the `x`, `y` are part of a module being `github.com/org/myProject`. The module name should be referenced in `cue.mod/module.cue`.

Thanks for your answer, now I understand the defination of Import path.But there is still a question about cue IDEA plugin,
in the cue.bnf, ImportPath       ::= "\"" ImportLocation [ ":" identifier ] "\""
but when I use import "github.com/org/myProject/packageX:x", the ":x" is not parsed as [ ":" identifier ], but "github.com/org/myProject/packageX:x" is parsed as a whole ImportLocation.

**Comment by verdverm** [comment]
fwif, GitHub highlights the `X` and `x` in the following. This should be the same as VS Code

```cue
import (
  X "github.com/org/myProject/packageX:x"
)
```

Note, you can use __```cue__ to get highlighting
